doc/ch-convert_dest.rst

```diff
@@ -0,0 +1,179 @@
+    image reference; otherwise, a filesystem path.
+
+  :code:`OUT`
+    Descriptor for the output image.
+
+  :code:`OUT_ARG ...`
+    (Valid only for :code:`tar` and :code:`squash` output formats.) Additional
+    arguments passed unchanged to the packing command.
+
+  :code:`-b`, :code:`--builder BUILDER`
+    Use image builder :code:`BUILDER`. Default: :code:`$CH_BUILDER` if set;
+    otherwise guessed (see :code:`ch-build(1)`).
+
+  :code:`-c`, :code:`--out-compress (none|gzip|xz)`
+    (Valid only for :code:`tar` and :code:`squash` output formats.)
+    Compression algorithm for archive file. Default: :code:`$CH_COMPRESSION`
+    if set; otherwise :code:`gzip`.
+
+  :code:`-h`, :code:`--help`
+    Print help and exit.
+
+  :code:`-i`, :code:`--in-fmt (builder|dir|tar|squash)`
+    Format of the input image. If omitted, inferred as described below.
+
+  :code:`--no-clobber`
+    Complain and fail if :code:`OUT` already exists, rather than replacing it.
+
+  :code:`-o`, :code:`--out-fmt (builder|dir|tar|squash)`
+    Format of the output image; inferred if omitted.
+
+  :code:`-s`, :code:`--builder-storage ARG`
+    Some builders take an argument specifying where to find their storage,
+    e.g. :code:`ch-grow --storage=/path`. If specified, pass :code:`ARG` to
+    the builder. Default: :code:`$CH_GROW_STORAGE` if set; otherwise, see
+    :code:`ch-grow(1)`. (Currently only implemented for :code:`ch-grow(1)`.)
+
+  :code:`--tmp DIR`
+    Path to temporary directory. Default: :code:`$TMP` if specified; otherwise
+    :code:`/tmp`.
+
+  :code:`-v`, :code:`--verbose`
+    Print extra chatter.
+
+
+Format and filename inference
+=============================
+
+:code:`ch-convert` tries to save typing by guessing formats and filenames when
+they are reasonably clear.
+
+Format inference is done according for both :code:`IN` and :code:`OUT`. The
+algorithm tries to match the value agains the following globs in this order.
+Paths need not exist in the filesystem.
+
+  1. :code:`*.sqfs`, :code:`*.squash`, :code:`*.squashfs`: SquashFS.
+  2. :code:`*.tar`, :code:`*.t?z`, :code:`*.tar.??`, :code:`*.tar.??`: tarball.
+  3. :code:`*:*` (i.e., containing a colon): builder storage.
+  4. Otherwise: directory.
+
+A notable consequence of these rules is than a simple image reference such as
+:code:`debian` is inferred to be a directory. The workaround is to add the
+default tag, e.g. :code:`debian:latest`.
+
+:code:`ch-convert` also infers filenames for :code:`OUT` (but not :code:`IN`)
+to save typing. If :code:`OUT` is path to a directory that exists, and the
+output format is:
+
+  * directory,
+    * and :code:`OUT` ends in slash: append inferred filename.
+    * otherwise: use :code:`OUT` without modification.
+  * tarball or squash: append inferred filename.
+  * builder storage: use :code:`OUT` without modification.
+
+Inferred filename rules: FIXME
+
+
+Supported conversions
+=====================
+
+Not all input/output format pairs are supported.
```

Comment:
This statement that not all input/output format pairs are supported, but the diagram states that every input/output format is either supported or will eventually be supported. Am I misinterpreting this?